fuel-p2p/src/gossipsub/builder.rs

```diff
@@ -1,31 +1,29 @@
-use libp2p::{
-    gossipsub::{
-        FastMessageId,
-        Gossipsub,
-        GossipsubConfigBuilder,
-        GossipsubMessage,
-        MessageAuthenticity,
-        MessageId,
-        PeerScoreParams,
-        PeerScoreThresholds,
-        RawGossipsubMessage,
-    },
-    identity::Keypair,
+use libp2p::gossipsub::{
+    FastMessageId,
+    Gossipsub,
+    GossipsubConfigBuilder,
+    GossipsubMessage,
+    MessageAuthenticity,
+    MessageId,
+    PeerScoreParams,
+    PeerScoreThresholds,
+    RawGossipsubMessage,
 };
 use sha2::{
     Digest,
+    Sha224,
     Sha256,
 };
 
 use crate::config::P2PConfig;
 
-pub fn build_gossipsub(local_key: &Keypair, p2p_config: &P2PConfig) -> Gossipsub {
+pub fn build_gossipsub(p2p_config: &P2PConfig) -> Gossipsub {
     let gossip_message_id = move |message: &GossipsubMessage| {
         MessageId::from(&Sha256::digest(&message.data)[..20])
     };
 
     let fast_gossip_message_id = move |message: &RawGossipsubMessage| {
-        FastMessageId::from(&Sha256::digest(&message.data)[..8])
+        FastMessageId::from(&Sha224::digest(&message.data)[..16])
```

Comment:
Since these algorithms will just pad the input out to some multiple of 256 or 224bits this is still doing roughly the same amount of work.

From reading the through the libp2p code, it seems that the fast message ID needs to be unique as well. This is because it creates a 1:1 mapping of fast message ID's to regular message IDs, so if there is a collision between fast message ID's, it will get marked as a duplicate even if the regular message ID is different. It seems the only difference for fast message ids is that they are computed before the message is decompressed, allowing for duplicate checks to skip message decoding / decompression. However, I think we should still probably hash the entire compressed message with something like a sha256 since the uniqueness matters here.